ssh.go

```diff
@@ -42,6 +42,9 @@ type PasswordHandler func(ctx Context, password string) bool
 // PtyCallback is a hook for allowing PTY sessions.
 type PtyCallback func(ctx Context, pty Pty) bool
 
+// LocalPortForwardingCallback is a hook for allowing port forwarding
+type LocalPortForwardingCallback func(ctx Context, destinationHost string, destinationPort uint32) bool
```

Comment:
I chose plain parameters here rather than a typed struct, mostly because I'm unhappy with the names - and it's easier to change parameter names than struct field names and I don't envision more parameters so far.  Happy to change it if I get definitive names.